### PR TITLE
Adding function scitoken_get_claim_string implementation

### PR DESCRIPTION
**Pull request by djw8605**
The SciToken object now stores claims in a map, rather than adding them directly to the builder.  Only create the builder when it is going to be signed and returned.

### CODE REVIEW

**Comment by bbockelm** [comment]
Can you separate out the UUID piece?  That definitely looks good (other than perhaps just utilizing `pkg-config` on Unix)?

For the `scitokens_get_claim_string` implementation - my original intent was to have this come from the `jwt::decoded` object, meaning that claim strings were only available once the token was verified.

Why do you want to do it on build side?  I suppose it'll work to do this approach, as long as we start with checking that the `jwt::decoded` member exists first (preferentially querying that), then falling back to the builder.

Thoughts?

**Comment by djw8605** [comment]
UUID separated. #6 

In the current implementation:
- You can set claims, but can't "get" them because they go directly into the `builder`.
- If it's a parsed token, you can only get old claims.

I think it may be necessary to parse all the claims into a structure for each deserialize.  And then put them in all in the `builder` for serialize.

File: src/scitokens_internal.cpp [hunk 1]
```diff
@@ -312,6 +312,9 @@ void
 SciToken::deserialize(const std::string &data, const std::vector<std::string> allowed_issuers) {
     m_decoded.reset(new jwt::decoded_jwt(data));
 
+    // Set all the claims
+    m_claims = m_decoded->get_payload_claims();
```
Comment (bbockelm) [inline src/scitokens_internal.cpp]: Can you move this to the end of the function?

That way, we only make a copy of the claims if the validator passes.

File: src/scitokens_internal.h [hunk 2]
```diff
@@ -159,6 +159,7 @@ friend class scitokens::Validator;
             builder.set_payload_claim(it.first, it.second);
         }
 
+        // TODO: handle JTI
```
Comment (bbockelm) [inline src/scitokens_internal.h]: What is the TODO here?

File: src/scitokens_internal.h [hunk 3]
```diff
@@ -111,39 +112,55 @@ friend class scitokens::Validator;
...
     }
 
+    const jwt::claim
+    get_claim(const std::string &key) {
+        return m_claims[key];
+    }
+
+    const std::string
+    get_claim_string(const std::string &key) {
+        return m_claims[key].as_string();
```
Comment (bbockelm) [inline src/scitokens_internal.h]: This can throw an exception if the claim is not a string, right?  Let's note it in the function comments.

**Pull request merged**